ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -1215,29 +1219,42 @@ knownIntersectionStateTop cfgEnv dynEnv intEnv =
 
     offerJump ::
         MkPipelineDecision
-     -> JumpInfo blk
+     -> Either (AnchoredFragment (Header blk)) (JumpInfo blk)
```

Comment:
What about extracting this as a separate type, eg
```haskell
data Instruction blk
  = RunNormally
  | JumpInstruction !(JumpInstruction blk)

data JumpInstruction blk =
  | JumpTo !(JumpInfo blk)
  | JumpToGoodPoint !(AF.AnchoredFragment (Header blk))
```
and use `JumpInstruction` here?

Reply:
Adopted.